docs/adr/0030-disable-squash-and-merge.md

```diff
@@ -0,0 +1,47 @@
+
+1. Accidentally squashing and merging train branches into `main` causes merge conflicts for the next patch release because of differing parent commits
+1. Accidentally forgetting to edit the commit message when squashing and merging an approved PR means PRs can be merged without meeting our commit guidelines
+1. Accidentally regular merging a branch that was meant to be squashed and merged means multiple unformatted commits can land for a single task
+1. While squashing and merging Dependabot (`package.json`) and Bananafox (`yarn.lock`) commits on automated dependency PRs seemed like a nice _pro_ for keeping squash and merge, there's been a couple of instances where something funky happened with a bad squash and merge dependency update which caused Bananafox to get out of sync and commit on multiple PRs without `package.json` changes. This caused an issue with a dot release that could have been mitigated by being able to cherry-pick in only the needed change without the `yarn.lock` commit, see [this PR](https://github.com/mozilla/fxa/pull/12064) and and others around this time, and so it seems more ideal to keep these in separate commits until Dependabot [works properly with yarn 2](https://github.com/dependabot/dependabot-core/issues/1297) so we can retire Bananafox
+
+## Considered Options
+
+- Continue allowing squash and merge through GH
+- Disable squash and merge
+
+## Decision Outcome
+
+Chosen option: "option 2", because with our current workflow, team discussions have concluded there doesn't appear to be adequate workarounds to alleviate the problems listed above while leaving squash and merge enabled. This is not a final decision; we will continue to have conversations around development strategy process improvements and viability of squash and merge in the future.
+
+### Positive Consequences
+
+- Train branches will no longer be accidentally squashed and merged, reducing merge conflicts and burden on release owners
+- While there is an extra step to squashing commits locally, pushing, and regular merging through GH rather than doing it all at once with a squash and merge button in GH, this extra step allows us to see exactly what's going to be merged when we click "merge." The 2nd and 3rd problems laid out above should be alleviated
+- Commit messages can be reviewed as part of the PR review. We don't need to have a hard rule for this, but it's oftentimes nicer reviewing a PR that has the type/scope/summary and commit message body filled out already, rather than r+'ing a PR with a bunch of "WIP" commits and trusting that it will be squashed into one commit that follows guidelines once the PR is merged
+- Better preservation of commit signing integrity. When we squash and merge through GH, GH signs the commit for us
+
+### Negative Consequences
+
+- Squashing and merging through GH can be convenient for engineers
+- Interactive rebase to squash and then force push a branch can feel scary since it overrides history
```

Comment:
I believe force push to main is disabled and we rarely if ever force push to train branches to help allievate some of the scariness.

Reply:
Yeah in the next paragraph I've got this...

> We also have branch protection against force pushing on `main`, and can add this protection to train branches.

It's currently disabled on `main`. It looks like we can disable it on `train` branches if we want to as well, which seems like a good idea to me.